dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClaimedTaskRestRepository.java

```diff
@@ -125,6 +135,54 @@ public ClaimedTaskRest findOne(Context context, Integer id) {
         }
     }
 
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @SearchRestMethod(name = "findAllByItem")
+    public Page<ClaimedTaskRest> findAllByItem(@Parameter(value = "uuid", required = true) UUID itemUUID,
+           Pageable pageable) {
+        List<ClaimedTask> tasks = null;
+        try {
+            Context context = obtainContext();
+            Item item = itemService.find(context, itemUUID);
+            if (item == null) {
+                return null;
```

Comment:
Both an incorrect item and no claim tasks for the items result in a 404, the incorrect item should most likely be a 429
Same applies for the other searches below

Reply:
@benbosman : A 429 is "Too Many Requests".  I don't think that's the error code we'd want here.  Is there a reason why 404 isn't sufficient?  I'll look at other search endpoints to see what they return in this scenario.

Reply:
@benbosman : We are returning 404s on (at least some) other endpoints of this type.  For example

* `/api/config/workflowdefinitions/search/findByCollection?uuid=[invalidCollection]` : https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkflowDefinitionRestRepositoryIT.java#L256
* `/api/eperson/registration/search/findByToken?token=[invalidToken]`: https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/test/java/org/dspace/app/rest/RegistrationRestRepositoryIT.java#L94

In my opinion, a 404 is sufficient here, unless I'm missing your point.

Reply:
Sorry, my mistake.
I meant 422, in line with https://github.com/DSpace/Rest7Contract/blob/main/vocabularies.md#get-controlled-vocabulary-by-metadata-and-collection

Reply:
Thanks for clarifying.  I'm OK with switching this to 422... I can see how that might give more information to the client. A 404 also is returned if the endpoint is "Not Found", so in this scenario I can understand that a 404 might be misleading -- the request is *valid*, but the UUID param is not.

@abollini : Do you have a strong opinion here?  I can see both sides, and it seems like we've been inconsistent on whether to return 422 or 404 in this scenario.

Reply:
thanks @benbosman @tdonohue to note that. `404` is not appropriate because the endpoint exists, it is just a matter of the result that must be returned or the parameters that have been provided.
There are four possible scenario here
- the item UUID is missing, this should lead to a 400 error as it is a mandatory parameter
- the parameter for the item UUID is invalid (it is a string, an integer, etc.) it should be treated as the previous case 400
- the item is not found, this should return 422
- the item exists but there are no claimed tasks for it, this should return an empty page with a 200 (as there is at least the pagination information in the json, if it was a single element search 204 would be used)

we have already had a detailed discussion around both aspects 404 vs 400 vs 422 and about the 200/204 for empty search but I was able to find previous references only for the first see
https://jira.lyrasis.org/browse/DS-4428
https://github.com/DSpace/Rest7Contract/pull/102

This also mean that the two enpoints noted by @tdonohue that currently return 404 are wrong and should be fixed

Reply:
@benbosman when a single result is returned by a search method, null is translated to 204 No Content. This is already supported by our RestResourceController, see https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/main/java/org/dspace/app/rest/RestResourceController.java#L1014

Reply:
I have also created https://github.com/DSpace/DSpace/issues/3093 to track the discussion about the other two endpoints